test/distributed/tensor/parallel/test_tp_style.py

```diff
@@ -229,15 +235,41 @@ def test_colwise_parallel_style(self):
         tensor = torch.rand(8, 16, device=self.device_type)
         cs = ColwiseParallel()
         self._1d_input_func_check(
-            tensor,
+            [tensor],
             tensor,
             cs._prepare_input,
+            error_msgs="No device mesh is currently active",
         )
         output, dtensor, device_mesh = self._test_prepare_output(
             cs._prepare_output, [Shard(-1)]
         )
         self.assertEqual(output, dtensor.to_local())
 
+    @with_comms
+    def test_prepare_module_input(self):
+        tensor = torch.rand(8, 16, device=self.device_type)
+        gathered_tensors = [
+            torch.empty_like(tensor) for _ in range(self.world_size)
+        ]
+        dist.all_gather(gathered_tensors, tensor)
+        gathered_tensors = torch.cat(gathered_tensors, dim=0).contiguous()
```

Comment:
nit: can just use functional collective without manually recreate the gathered tensors. Can fix this later

Reply:
oh ok, sure. Will do in follow up PR which I will try to change test.